packages/react-component-library/src/components/Nav/Nav.test.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { shallow, ShallowWrapper } from 'enzyme'
+import renderer from 'react-test-renderer'
+
+import Nav from './index'
+
+describe('Nav', () => {
+  let component: ShallowWrapper
+  let navItems: any[]
+  let navItemsWithChildren: any[]
+
+  navItems = [
+    {
+      href: 'http://testurl.test',
+      label: 'Styles',
+    },
+    {
+      href: 'http://testurl.test',
+      label: 'Components',
+    },
+    {
+      href: 'http://testurl.test',
+      label: 'Patterns',
+      active: true,
+    },
+    {
+      href: 'http://testurl.test',
+      label: 'Community',
+    },
+    {
+      href: 'http://testurl.test',
+      label: 'About',
+    },
+  ]
+
+  navItemsWithChildren = [
+    ...navItems,
+    {
+      href: 'http://testurl.test',
+      label: 'Parent',
+      children: [
+        {
+          href: 'http://testurl.test',
+          label: 'Child 1',
+        },
+        {
+          href: 'http://testurl.test',
+          label: 'Child 2',
+        },
+      ]
+    },
+  ]
+
+  describe('Given the nav is supplied navItems prop', () => {
```

Comment:
We can talk more about agreed test methods later. But I'd be tempted to not just test the render but inspect what is rendered in individual tests

```
describe('Nav' => () ={
  describe('Given the nav is supplied a flat navItems collection', () => {
    let navItems

    beforeEach(() => {
      navItems = ....
    })

    describe('and is asked to render vertically', () => {
      let wrapper

      beforeEach(() => {
        wrapper =shallow(<Nav navItems={navItems} orientation="vertical"></Nav>)
      })

     it('should indicate that the navigation is vertical', () => {
       expect(wrapper.class).toInclude('vertical')';
     })

    it('should render 5 list items', () => {
...
})
```

That idea being that you nest tests and build up contexts and then each it test just 1 thing, or one aspect.

Reply:
Sounds good to me I like this style.

Reply:
I'm currently checking the appropriate change to the rendered output happens for the different potential variants of each prop.

Then i'd do the same for user input (if it was an interactive component) and ensure that appropriate events are emitted or any expected side effects are initiated.